### ISSUE TITLE
[Bug] TypeError: root.stringify is not a function

### ISSUE THREAD

**Issue by constgen** [gucong3000/postcss-html#7]
I get the next exception when try to preload an HTML file with PostCSS

```bash
× Building project failed
./src/svelte/main.html
Module build failed: TypeError: root.stringify is not a function
    at document.nodes.forEach.root (D:\Work\Projects_Fabrica\neutrino-preset-fabrika\node_modules\postcss-html\lib\stringify.js:10:9)
    at Array.forEach (native)
    at Function.stringify (D:\Work\Projects_Fabrica\neutrino-preset-fabrika\node_modules\postcss-html\lib\stringify.js:7:18)
    at MapGenerator.generate (D:\Work\Projects_Fabrica\neutrino-preset-fabrika\node_modules\postcss\lib\map-generator.js:311:18)
    at LazyResult.stringify (D:\Work\Projects_Fabrica\neutrino-preset-fabrika\node_modules\postcss\lib\lazy-result.js:290:24)
    at Root.toResult (D:\Work\Projects_Fabrica\neutrino-preset-fabrika\node_modules\postcss\lib\root.js:105:21)
    at lintPostcssResult (D:\Work\Projects_Fabrica\neutrino-preset-fabrika\node_modules\stylelint\lib\lintSource.js:107:6)
    at getConfig.then.result (D:\Work\Projects_Fabrica\neutrino-preset-fabrika\node_modules\stylelint\lib\lintSource.js:75:16)
 @ ./src/index.js 6:0-38
 @ multi ./src/index
```

I inspected the `root` object in `stringify.js` and looks like it doesn't have such method. It inherits from these interfaces:

```
Rule {}

Container {
  push: [Function: push],
  each: [Function: each],
  walk: [Function: walk],
  walkDecls: [Function: walkDecls],
  walkRules: [Function: walkRules],
  walkAtRules: [Function: walkAtRules],
  walkComments: [Function: walkComments],
  append: [Function: append],
  prepend: [Function: prepend],
  cleanRaws: [Function: cleanRaws],
  insertBefore: [Function: insertBefore],
  insertAfter: [Function: insertAfter],
  removeChild: [Function: removeChild],
  removeAll: [Function: removeAll],
  replaceValues: [Function: replaceValues],
  every: [Function: every],
  some: [Function: some],
  index: [Function: index],
  normalize: [Function: normalize],
  rebuild: [Function: rebuild] }

Node {
  error: [Function: error],
  warn: [Function: warn],
  remove: [Function: remove],
  toString: [Function: toString],
  clone: [Function: clone],
  cloneBefore: [Function: cloneBefore],
  cloneAfter: [Function: cloneAfter],
  replaceWith: [Function: replaceWith],
  moveTo: [Function: moveTo],
  moveBefore: [Function: moveBefore],
  moveAfter: [Function: moveAfter],
  next: [Function: next],
  prev: [Function: prev],
  before: [Function: before],
  after: [Function: after],
  toJSON: [Function: toJSON],
  raw: [Function: raw],
  root: [Function: root],
  cleanRaws: [Function: cleanRaws],
  positionInside: [Function: positionInside],
  positionBy: [Function: positionBy] }
```

Is this module compatible with PostCSS 6?

**Comment by gucong3000** [comment]
> Is this module compatible with PostCSS 6?

Yes, see: https://github.com/gucong3000/postcss-html/blob/master/package.json#L67

**Comment by gucong3000** [comment]
@constgen 
Can you push a PR to reproduce the problem?
Or upload a zip pack?

**Comment by kapiljaveri** [comment]
Hello,

I seem to be running into the same issue as well. It seems as though it fails in the stringify function when the root node in the forEach isn't of type: 'root' but instead root: 'rule' or type: 'atrule'

I see this when I include postcss-import or css-mqpacker..
if that helps...

**Comment by gucong3000** [comment]
@constgen
@kapiljaveri 
Please use v0.9.0, if this issue persists, please upload your stylelint/postcss configuration and HTML.

**Comment by constgen** [comment]
I would like to provide a test suite. But infrastructure is too complex and it is built on linked packages. So it would not be possible to reproduce. 
Thanks. I will try the latest version. All previos updates didn't work.

**Comment by kapiljaveri** [comment]
@gucong3000 

Sorry it took long, but attached is a simple example of where it is failing. The only postcss plugin it is using is css-mqpacker and like I mentioned earlier, it seems to be failing on 'atrules'.
[postcss-html-test.zip](https://github.com/gucong3000/postcss-html/files/1458970/postcss-html-test.zip)

You should be able to run `npm intall` and then `npm test` too have it give you what you need.

Let me know if you need anything else or if you think I'm going something wrong.

Thanks!

**Comment by gucong3000** [comment]
@kapiljaveri 
Thanks, with your help, I found the problem.

**Comment by gucong3000** [comment]
fixed in v0.11.0

**Comment by kapiljaveri** [comment]
@gucong3000 
Got it and it worked! Thank you.

I did find another issue related to some of the changes you checked in around  [document.append / prepend](https://github.com/gucong3000/postcss-html/commit/6f252374c8ff0d10dc8317b76b5515add1d2397f#diff-3fab227e34d65fe9bb2d6ea53eec41cfR59)

It complains about [this.last](https://github.com/gucong3000/postcss-html/commit/6f252374c8ff0d10dc8317b76b5515add1d2397f#diff-3fab227e34d65fe9bb2d6ea53eec41cfR60) being undefined hence causing the append to fail. If you'd like I can create a new issue with sample code?

Thanks again on the quick resolution for the current issue!

**Comment by gucong3000** [comment]
> If you'd like I can create a new issue with sample code?

Of course, please.
